logic/src/commonTest/kotlin/com/wire/kalium/logic/data/user/UserRepositoryTest.kt

```diff
@@ -198,6 +199,22 @@ class UserRepositoryTest {
             .wasNotInvoked()
     }
 
+    @Test
+    fun givenAnUserIdListWithDifferentDomain_whenApiReturnsFederationDisabledError_thenShouldTryToFetchOnlyUsersWithSelfDomain() = runTest {
+        // given
+        val requestedUserIds = setOf(TestUser.OTHER_USER_ID, TestUser.OTHER_USER_ID_2)
+        val (arrangement, userRepository) = Arrangement()
+            .withGetMultipleUsersApiRequestFederationNotEnabledError()
+            .arrange()
+        // when
+        userRepository.fetchUsersByIds(requestedUserIds).shouldFail()
+        // then
+        verify(arrangement.userDetailsApi)
+            .suspendFunction(arrangement.userDetailsApi::getMultipleUsers)
+            .with(any())
+            .wasInvoked(exactly = twice)
```

Comment:
It would be nice to check that it invoked `getMultipleUsers` with the different inputs, instead of `any()` tiwce.